dotnet-engine/Yggdrasil.Engine/NativeLoader.cs

```diff
@@ -0,0 +1,179 @@
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+internal static class NativeLibLoader
+{
+    internal static IntPtr LoadNativeLibrary()
+    {
+        var libName = GetBinaryName();
+        var tempPath = Path.Combine(Path.GetTempPath(), libName);
+        var assembly = Assembly.GetExecutingAssembly();
+        var assemblyName = assembly.GetName().Name;
+
+        if (!File.Exists(tempPath))
```

Comment:
Stolen from Java - we extract the binary to a temp folder and then use built in tools to mount it. Gotta do this because .NET framework won't respect binaries in the nuget package itself